### PR TITLE
Revamping edit-with-emacs feature

### PR DESCRIPTION
**Pull request by agzam**
I'm revamping and cleaning up 'edit-with-emacs' feature.

It needs to be simpler. Currently, it allows you to edit in one app at a time, and that's rather annoying.

I am going to remove frame manipulation - whenever the edit session gets initiated, there will be no new frame by default. All frame/window manipulations should be on the Emacs side, without Hammerspoon - using `display-buffer-alist` and the hooks.

This probably will be a breaking change for some, but I'll try to make the transition as smooth as possible.

### CODE REVIEW

File: docs/emacs.org [hunk 1]
```diff
@@ -2,34 +2,124 @@
 
   [[./edit-with-emacs-demo.gif/]]
 
-  Any text, in just any app can be edited using Emacs. The idea is simple - you press a dedicated key-combo (default: =Cmd+Ctrl+O=), Spacehammer copies the existing text & calls ~emacsclient~, then it invokes a function that opens a buffer and pastes the text into it. After you're done editing the text in Emacs, you press =C-c C-c=, it grabs the text, switches back to the program & pastes the text back in there. It works surprisingly well.
+  Any text, in just any app can be edited using Emacs. The idea is simple - you press a dedicated key-combo (default: =Cmd+Ctrl+O=), Spacehammer copies the existing text & calls ~emacsclient~, which invokes a function that opens a buffer and pastes the text into it. Once you finish editing, you press =C-c C-c=, Emacs grabs the text, switches back to where you were before and pastes the new text in there. It works surprisingly well.
```
Comment (Grazfather) [inline docs/emacs.org:5]: But: what do you mean by "just" any app. You mean "just about any app" or "any app". "Just any app" sounds strange.

File: docs/emacs.org [hunk 2]
```diff
@@ -2,34 +2,124 @@
...
+
+#+begin_src elisp
+(when (eq system-type 'darwin)
+  (package! spacehammer :recipe (:host github
+                                 :repo "agzam/spacehammer"
+                                 :files ("*.el"))))
+#+end_src
+
+***** Or symlink to the folder:
+Since you already have the package file in ~/.hammerspoon/, instead of loading it from GitHub, you may choose to load it directly.
```
Comment (Grazfather) [inline docs/emacs.org:42]: Probably the better idea so that they remain in sync?

File: emacs.fnl [hunk 3]
```diff
@@ -20,38 +27,27 @@
         title       (.. "\"" (: current-app :title) "\"")
         screen      (.. "\"" (: (hs.screen.mainScreen) :id) "\"")
         run-str     (..
-                     "/usr/local/bin/emacsclient"
-                     " -c -F '(quote (name . \"edit\"))' "
-                     " -e '(spacehammer/edit-with-emacs "
+                     (emacsclient-exe)
+                     ;; " -c -F '(quote (name . \"edit\"))' "
```
Comment (Grazfather) [inline emacs.fnl:31]: Remove?

File: spacehammer.el [hunk 4]
```diff
@@ -1,154 +1,139 @@
...
-      (global-edit-with-emacs-mode 1))
+(defvar spacehammer-edit-with-emacs-hook nil
+  "Hook for when edit-with-emacs buffer gets activated.
+Hook function must accept arguments:
+- `buffer-name' - the name of the edit buffer
+- `pid'         - PID of the app that invoked Edit-with-Emacs
+- `title'       - title of the app that invoked Edit-with-Emacs")
+
+(defvar spacehammer-before-finish-edit-with-emacs-hook nil
+  "`edit-with-emacs' finished and dedicated buffer and frame about to get deleted.
```
Comment (Grazfather) [inline spacehammer.el:58]: Nit: wording is a bit strange

File: spacehammer.el [hunk 5]
```diff
@@ -1,154 +1,139 @@
...
+Hook function must accept arguments:
+- `buffer-name' - the name of the edit buffer
+- `pid'         - PID of the app that invoked Edit-with-Emacs")
+
+(defvar spacehammer-before-cancel-edit-with-emacs-hook nil
+  "`edit-with-emacs' canceled and dedicated buffer and frame about to get deleted.
```
Comment (Grazfather) [inline spacehammer.el:64]: Same wording nit

File: spacehammer.el [hunk 6]
```diff
@@ -1,154 +1,139 @@
...
-
-(define-global-minor-mode global-edit-with-emacs-mode
-  spacehammer/edit-with-emacs-mode spacehammer/turn-on-edit-with-emacs-mode)
-
-(defvar systemwide-edit-previous-app-pid nil
-  "Last app that invokes `spacehammer/edit-with-emacs'.")
-
-(defun spacehammer/finish-edit-with-emacs ()
+(defun spacehammer-finish-edit-with-emacs ()
+  "When done editing."
```
Comment (Grazfather) [inline spacehammer.el:100]: What does this do?

**Review by Grazfather — APPROVED** [review]
I don't know elisp or emacs well. Read through and it makes sense. Tried to focus on wording since it's truly new code to me. Let me know if you'd like me to test it out with doom.

**Comment by agzam** [comment]
@Grazfather Great review, thank you! Please check the "PR feedback" commit. Let me know if you spot anything else.

**Review by Grazfather — APPROVED** [review]

**Pull request merged**